streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java

```diff
@@ -67,6 +68,68 @@
     }
 
     @Override
+    public KeyValueIterator<Bytes, byte[]> all() {
+        final List<Segment> searchSpace = segments.allSegments();
+        
+        final Segment minSegment = segments.minSegment();
+        final Segment maxSegment = segments.maxSegment();
+        final Bytes minKey = minSegment.first().key;
+        final Bytes maxKey = maxSegment.last().key;
+        
+        final long minTimestamp = keySchema.segmentTimestamp(minKey);
+        final long maxTimestamp = keySchema.segmentTimestamp(maxKey);
+        final Bytes keyFrom = WindowStoreUtils.bytesKeyFromBinaryKey(minKey.get());
+        final Bytes keyTo = WindowStoreUtils.bytesKeyFromBinaryKey(maxKey.get());
+        
+        return new SegmentIterator(searchSpace.iterator(),
+                                   keySchema.hasNextCondition(keyFrom, keyTo, minTimestamp, maxTimestamp),
+                                   minKey, maxKey);
+    }
+    
+    @Override
+    public KeyValueIterator<Bytes, byte[]> fetchAll(final long timeFrom, final long timeTo) {
+        final Segment minSegment = segments.getMinSegmentGreaterThanEqualToTimestamp(timeFrom);
+        final Segment maxSegment = segments.getMaxSegmentLessThanEqualToTimestamp(timeTo);
+        
+        long minTimestamp = Long.MAX_VALUE;
+        long maxTimestamp = 0;
+        Bytes keyFrom = new Bytes(new byte[0]);
+        Bytes keyTo = null;
+        
+        final KeyValueIterator<Bytes, byte[]> iteratorMin = minSegment.all();
+        while (iteratorMin.hasNext()) {
+            final KeyValue<Bytes, byte[]> bytes = iteratorMin.next();
+            final long timestamp = keySchema.segmentTimestamp(bytes.key);
+            
+            if (timestamp >= timeFrom && timestamp <= timeTo) {
+                final Bytes currentBytes = WindowStoreUtils.bytesKeyFromBinaryKey(bytes.key.get());
+                if (timestamp < minTimestamp) {
+                    minTimestamp = timestamp;
+                    keyFrom = currentBytes;
+                } else {
+                    keyFrom = keyFrom.compareTo(currentBytes) > 0 ? currentBytes : keyFrom;
+                }
+            }
+        }
+        final KeyValueIterator<Bytes, byte[]> iteratorMax = maxSegment.all();
+        while (iteratorMax.hasNext()) {
+            final KeyValue<Bytes, byte[]> bytes = iteratorMax.next();
+            final long timestamp = keySchema.segmentTimestamp(bytes.key);
+            
+            if (timestamp >= timeFrom && timestamp <= timeTo) {
+                final Bytes currentBytes = WindowStoreUtils.bytesKeyFromBinaryKey(bytes.key.get());
+                if (timestamp > maxTimestamp) {
+                    maxTimestamp = timestamp;
+                    keyTo = currentBytes;
+                } else {
+                    keyTo = keyTo.compareTo(currentBytes) < 0 ? currentBytes : keyTo;
```

Comment:
`keyTo` could be null here i think?  so the `compareTo` might throw an NPE

Reply:
There shouldn't be any chance of that happening, since the accompanying ```if``` clause for this ```else``` statement would reassign ```keyTo``` to ```currentBytes``` when ```currentTimestamp``` is bigger. This should always be the case, especially since ```maxTimestamp``` was originally set to ```0`.